### PR TITLE
Add Metric Benchmark tests to integration tests

### PR DESCRIPTION
**Pull request by TheNiron**
Benchmark test are added for the metrices No. of SQLs executed and No. of external API calls for the following scenarios.

1. Creating an API
2. Publishing an API
3. Retrieving all APIs from publisher (page 1)
4. Retrieving all APIs from store (page 1)
5. Retrieving an API from publisher
6. Retrieving an API from store
7. Creating an Application
8. Subscribing to an API
9. Generating an access token - JWT
10. Invoking the created API with JWT
11. Create an API Revision And Deploy

- The benchmark-tests module is placed under tests-integration, and will be executed before tests-backend module
- The logs are read from correlation logs.
- Benchmark values are defined in "src/test/resources/benchmark-values"
- For each test scenario relevant logs will be extracted and can be found in "target/logs/benchmark-tests"

### CODE REVIEW

File: modules/integration/tests-integration/tests-benchmark/src/test/java/org/wso2/am/integration/tests/benchmarktest/BenchmarkTestCase.java [hunk 1]
```diff
@@ -0,0 +1,409 @@
...
+        HttpResponse response = restAPIPublisher
+                .changeAPILifeCycleStatus(apiUUID, APILifeCycleAction.PUBLISH.getAction(), null);
+        assertEquals(response.getResponseCode(), Response.Status.OK.getStatusCode(),
+                "API publish Response code is invalid " + apiUUID);
+
+//        Validate the JDBC query counts executed from correlation log
+        benchmarkUtils.validateBenchmarkResults(testName, JDBC_METRIC, startTime, scenario, providerName);
+//        Validate external api request counts executed from correlation log
+        benchmarkUtils.validateBenchmarkResults(testName, EXTERNAL_API_METRIC, startTime, scenario, providerName);
+//        benchmarkUtils.deleteRestAPI(apiUUID);
```
Comment (bhathiya) [inline modules/integration/tests-integration/tests-benchmark/src/test/java/org/wso2/am/integration/tests/benchmarktest/BenchmarkTestCase.java:139]: please remove these unwanted lines.

Comment (TheRasanjana) [reply modules/integration/tests-integration/tests-benchmark/src/test/java/org/wso2/am/integration/tests/benchmarktest/BenchmarkTestCase.java:138]: This is done.

File: modules/integration/tests-integration/tests-benchmark/src/test/java/org/wso2/am/integration/tests/benchmarktest/BenchmarkTestCase.java [hunk 2]
```diff
@@ -0,0 +1,397 @@
+/*
+ *Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (malinthaprasan) [inline modules/integration/tests-integration/tests-benchmark/src/test/java/org/wso2/am/integration/tests/benchmarktest/BenchmarkTestCase.java:2]: 2021

**Review by malinthaprasan — COMMENTED** [review]
As discussed:
- Lets try to reduce the copied artifacts from the existing tests 
    - There seems to be some files getting deployed due to `APIMCarbonServerExtension` getting executed via automation.xml. We can take a copy of that and remove all unnecessary deployments [1]. We can reconfigure the new class in automation.xml that you commit with this. 
- Let's no commit binary files unless absolutely necessary.
[1] https://github.com/wso2/product-apim/blob/e1a00ed5f0b36c9866f35dac5a0e2890def48c38/modules/integration/tests-common/framework-extensions/src/main/java/org/wso2/am/framework/extensions/APIMCarbonServerExtension.java#L103-L163

File: modules/integration/tests-integration/tests-benchmark/src/test/java/org/wso2/am/integration/tests/benchmarktest/BenchmarkUtils.java [hunk 3]
```diff
@@ -0,0 +1,251 @@
+/*
+ *Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (malinthaprasan) [inline modules/integration/tests-integration/tests-benchmark/src/test/java/org/wso2/am/integration/tests/benchmarktest/BenchmarkUtils.java:2]: 2021

File: modules/integration/tests-integration/tests-benchmark/src/test/java/org/wso2/am/integration/tests/api/lifecycle/APIManagerLifecycleBaseTest.java [hunk 4]
```diff
@@ -0,0 +1,678 @@
...
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.testng.Assert.assertEquals;
+
+/**
+ * Base test class for all API Manager lifecycle test cases. This class contents the all the
+ * common variables and t methods.
+ */
+public class APIManagerLifecycleBaseTest extends APIMIntegrationBaseTest {
```
Comment (malinthaprasan) [inline modules/integration/tests-integration/tests-benchmark/src/test/java/org/wso2/am/integration/tests/api/lifecycle/APIManagerLifecycleBaseTest.java:57]: Shall we reuse the one in existing tests without copying this class. This doesn't seem to contain any changes.

Comment (TheNiron) [reply modules/integration/tests-integration/tests-benchmark/src/test/java/org/wso2/am/integration/tests/api/lifecycle/APIManagerLifecycleBaseTest.java]: This is fixed

**Review by malinthaprasan — APPROVED** [review]

**Pull request merged**